chanzuckerberg/single-cell-data-portal
# feat: Curators can start a revision from the Collection page (#2647)

## Reason for Change

Curators can start a revision from the Collection page and can filter Collections by curator and status.

### Related Epics
- #2647
- #2638 

### Related Issues
- #4559 
- #4591
- #4598 
- #4604 
- #4603
- #4605 
- #4607
- #4555 

## Changes

- Curators can filter by curator and status as per mocks [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-43440&t=eneD3qBxvN1BSNdk-4) and [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-43446&t=eneD3qBxvN1BSNdk-4).
- Curators can start a revision from the Collection page as per mocks [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4461-39456&t=eneD3qBxvN1BSNdk-4) and view the revision as per mocks [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4532-39739&t=eneD3qBxvN1BSNdk-4) and continue the revision as per mocks [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4531-39402&t=eneD3qBxvN1BSNdk-4).
- The collection in revision can also be accessed from the Collections page from the collection row revision button as per mocks [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-40099&t=eneD3qBxvN1BSNdk-4).
- The new flow for curators, is to be enable curators to easily [create](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-40099&t=eneD3qBxvN1BSNdk-4) / [view](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4461-39456&t=eneD3qBxvN1BSNdk-4) / [edit](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4532-39739&t=eneD3qBxvN1BSNdk-4) / [continue revision](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4531-39402&t=eneD3qBxvN1BSNdk-4) / [publish](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4532-39739&t=eneD3qBxvN1BSNdk-4) collections, from either the Collections page or Collection detail page.
- The "My Collections" page is removed.
- The "Delete" collection button is removed see https://github.com/chanzuckerberg/single-cell-data-portal/issues/2099.
- Table sorting is added to `Curator` and `Status` columns only as per [mocks](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4846-40544&t=fu5LLOTZUeOuIydr-4).
- Updates to Collections / Datasets / Collection queries include adding an infinite stale time to minimise network requests, and where required, query invalidation to eliminate the need for manual page refresh in order to view changes.

## Testing steps

- Without authorisation: 
  - A user should only see the Collections "default" view mode e.g. Collections and Datasets should remain unchanged:
  - Collections should display `Publication`, `Tissue`, `Disease`, `Organism`.
  - Collection filters should include `Assay`, `Author`, `Cell Type`, `Development Stage`, `Disease`, `Organism`, `Publication Date`, `Self-Reported Ethnicity`, `Sex`, `Suspension Type` and `Tissue`.
- Curator:
  - Collections should also display `Status` and `Curator`, `Assay` and `Cells`.
  - Collections should not display `Publication`.
  - Collection filters should include `Status` and `Curator`, below `Tissue` and a horizontal divider.
  - A curator should have write access to the curator's own collections i.e. the curator should be able to:
    - create / view / edit / delete or publish a private collection,
    - view / start a revision on a published collection,
    - view / edit / publish or delete a private revision collection, and
    - continue a revision of a published collection.
  - A curator should not have visibility of other curator's private collections.
- Super Curator 
  - A super curator should have write access to all collections.

## Notes for Reviewer

- Tests are updated as required.

rdev: https://cc-rdev-frontend.rdev.single-cell.czi.technology/
